sample/src/main/kotlin/com/uber/autodispose/sample/ImageRepository.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2018. Uber Technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.uber.autodispose.sample
+
+import android.content.res.Resources
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import com.jakewharton.rx.replayingShare
+import com.jakewharton.rxrelay2.PublishRelay
+import io.reactivex.Observable
+
+/**
+ * Repository layer for loading a Bitmap from res/raw.
+ *  If you're using DI, this should be a singleton class,
+ *  so that [imageObservable] can be reused across
+ *  activities/fragments.
+ *
+ * @param resources
+ */
+class ImageRepository(private val resources: Resources) {
+
+  // Relay to accept the raw image ids that are to be loaded.
+  private val relay = PublishRelay.create<Int>()
+
+  /**
+   * The Observable returned to the UI. This will map each given
+   * resourceId to a Bitmap and cache it using [replayingShare].
+   *
+   * The caching is necessary since this is an expensive operation
+   * and should not be triggered every time someone subscribes to it.
+   * [replayingShare] helps with that.
+   */
+  private val imageObservable = relay
+      .map {
+        val imageStream = resources.openRawResource(it)
+        return@map BitmapFactory.decodeStream(imageStream)
```

Comment:
nit: let's clean this up a bit - `BitmapFactory.decodeStream(resources.openRawResource(it))`